src/java.base/share/classes/java/lang/constant/ConstantUtils.java

```diff
@@ -126,6 +126,10 @@ static String dropFirstAndLastChar(String s) {
         return s.substring(1, s.length() - 1);
     }
 
+    static boolean isPrimitiveArray(String referenceTypeDescriptorString) {
+        return referenceTypeDescriptorString.charAt(referenceTypeDescriptorString.length() - 1) != ';';
```

Comment:
it also needs to check if it starts with `[`


```suggestion
        return referenceTypeDescriptorString.startsWith("[") &&  
                  referenceTypeDescriptorString.charAt(referenceTypeDescriptorString.length() - 1) != ';';
```

Reply:
I intend this to be a `ReferenceClassDesc`-specifc check. I think moving `isPrimitiveArray` as a private method in `ReferenceClassDesc` would be better, as the `startsWith("[")` is already implied (at least before valhalla, non-array, i.e. class or interface descriptors must end with a semicolon.